tests/ui/const-generics/issues/issue-99641.rs

```diff
@@ -3,10 +3,10 @@
 
 fn main() {
     pub struct Color<const WHITE: (fn(),)>;
-    //~^ ERROR using function pointers
+    //~^ ERROR `(fn(),)` can't be used as a const parameter type
```

Comment:
Perhaps it would be useful to use the innermost failing `ConstParamTy` predicate here, so we mention `fn()` instead of `(fn(),)`... What do you think?

Reply:
Yeah that would make sense, we should probably also have a test case like:
```rust
#[derive(Eq, PartialEq)]
struct Foo<T>(T);

trait Other {}

impl ConstParamTy for Foo
where
    T: Other, {}

fn foo<const N: Foo<u8>>() {}
```
which should tell you about `u8: Other` not holding which is required for `Foo<u8>` to implement `ConstParamTy`

Reply:
Done